src/components/MessageInput.jsx

```diff
@@ -0,0 +1,107 @@
+/** @jsxImportSource @emotion/react */
+import { useState, useRef } from 'react';
+import { css } from '@emotion/react';
+import Search from '/public/assets/search.png';
+import Send from '/public/assets/send.png';
+
+const MessageInput = ({ onSendMessage }) => {
+  const [newMessage, setNewMessage] = useState('');
+  const textareaRef = useRef(null);
+
+  // form 제출 시 새로고침 방지
+  const handleFormClick = (e) => {
+    e.preventDefault();
+  };
+
+  const handleTextareaChange = (e) => {
+    setNewMessage(e.target.value);
+  };
+
+  const handleSendMessage = () => {
+    //공백 입력 방지
+    if (newMessage.trim() === '') return;
+
+    onSendMessage(newMessage); // 부모 컴포넌트로 새로운 메세지 전송
+    setNewMessage(''); // 입력 필드 초기화
+
+    // ref는 항상 존재여부를 검사하고 사용해야 한다. (단축평가)
+    textareaRef.current && textareaRef.current.focus();
+  };
+
+  const onCheckEnter = (e) => {
+    if (e.key === 'Enter' && e.nativeEvent.isComposing == false) {
+      e.preventDefault(); // Enter로 줄바꿈 방지, 메세지 전송
+      handleSendMessage();
+
+      setNewMessage('');
+      textareaRef.current && textareaRef.current.focus(); // 메세지 전송 후 textarea에 포커스
+    }
+  };
+
+  return (
+    <div>
+      <form
+        onSubmit={handleFormClick}
+        onKeyDown={onCheckEnter}
+        css={messageInputForm}
+      >
+        <button css={searchButton}>
+          <img src={Search} css={imageSize} />
+        </button>
+        <textarea
+          value={newMessage}
+          onChange={handleTextareaChange}
+          css={messageInputText}
+          placeholder="메세지 보내기..."
+          autoFocus
+        ></textarea>
```

Comment:
Child 요소가 없는 경우 self-closing 형식으로 쓸 수 있습니다!
```suggestion
        <textarea
          value={newMessage}
          onChange={handleTextareaChange}
          css={messageInputText}
          placeholder="메세지 보내기..."
          autoFocus
        />
```